src/SDDPoptimize.jl

```diff
@@ -292,6 +285,13 @@ function build_model(model, param, t)
         @objective(m, Min, cost + alpha)
     end
 
+    # Add binary variable if problem is a SMIP:
+    if model.IS_SMIP
+        m.colCat[1:nx] = model.stateCat
```

Comment:
This is not planned for the moment, but maybe in long term. That's why we add the possibility to define state as integer variables. Maybe we should add a warning if some users want to try this possibility ...